File: LibraryMe.API/BookLibrary.BAL/Services/Implementations/BookmarkService.cs
using BookLibrary.BAL.Services.Interfaces;
using BookLibrary.DAL.Repositories.Interfaces;
using BookLibrary.DAL.Models.DTO;

namespace BookLibrary.BAL.Services.Implementations
{
    public class BookmarkService : IBookmarkService
    {
        private readonly IBookmarkRepository _bookmarkRepo;

        public BookmarkService(IBookmarkRepository bookmarkRepo)
        {
            _bookmarkRepo = bookmarkRepo;
        }

        public async Task<List<BookmarkDTO>> GetBookmarksByUserId(Guid userId, string bookId = null)
        {
            return await _bookmarkRepo.GetBookmarksByUserId(userId, bookId);
        }

        public async Task<Guid> CreateBookmark(CreateBookmarkDTO dto)
        {
            return await _bookmarkRepo.CreateBookmark(dto);
        }

        public async Task<BookmarkDTO?> DeleteBookmark(Guid id)
        {
            return await _bookmarkRepo.DeleteBookmark(id);
        }
    }
}

File: LibraryMe.UI/LibraryMe/src/app/Features/Profile/services/bookmark-service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {CreateBookmarkDTO} from "../models/create-bookmark-dto";
import {Observable} from "rxjs";
import {BookmarkDTO} from "../models/bookmark-dto";
import {environment} from "../../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class BookmarkService {
  private baseUrl = 'https://localhost:7176/api/bookmarks';

  constructor(private httpClient: HttpClient) {}

  getBookmarksByUserId(userId: string, bookId:string=""): Observable<BookmarkDTO[]> {
    const url = `${environment.apiBaseUrl}/api/bookmarks/${userId}?bookId=${bookId}`;
    return this.httpClient.get<BookmarkDTO[]>(url);
  }

  createBookmark(dto: CreateBookmarkDTO): Observable<any> {
    return this.httpClient.post(`${environment.apiBaseUrl}/api/bookmarks`, dto);
  }

  deleteBookmark(id: string): Observable<BookmarkDTO> {
    const url = `${environment.apiBaseUrl}/api/bookmarks/${id}`;
    return this.httpClient.delete<BookmarkDTO>(url);
  }
}
